// components/actor/ActorCard.tsx
import React from 'react'
import notfound from "images/not-found.png"
import {StyledActorCard} from "./ActorCard.styled"

export type ActorProps = {
    person: {
        id: number,
        name: string,
        image: {
            medium: string,
            original:string,
        },
        country: {
            name:string
        }
        birthday: string,
        gender: string,
        deathday: string,
        voice: boolean,
        character: {
            name:string
        }

    }

}
const ActorCard = ({ person: { image, gender,name, country, birthday, deathday } }: ActorProps) => {
    const imgSrc = image ? image.medium : notfound

   return  <StyledActorCard>
      <div className='img-wrapper'>
        <img src={imgSrc} alt="actor" />
      </div>
      <h1>
        {name} {gender ? `(${gender})` : null}
      </h1>
      <p>{country ? `Comes from ${country.name}` : 'No country known'}</p>
      {birthday ? <p>Born {birthday}</p> : null}
      <p className='deathday'>{deathday ? `Died ${deathday}` : 'Alive'}</p>
    </StyledActorCard>
}

export default ActorCard
