// layout/MainPageLayout.tsx
import React, { useState } from 'react'
import {apiGet} from "misc/config"

type MainLayputProps = {
    children:React.ReactNode
}

type Show = {
    show: {
        id: number,
        name: string,
        type: string,
        language: string,
        genres: string[],
        rating: {
            average:number
        }
        image: {
            medium: string,
            original:string
        },
        summary:string
    }
}
type People = {
    person: {
        id: number,
        name: string,
        image: {
            medium: string,
            original:string
        },

    }

}

const MainPageLayout = ({ children }: MainLayputProps) => {
    const [input, setInput] = useState("")
    const [movies, setMovies] = useState<Show[] | People []>([])
    const [searchOption, setSearchOption] = useState('shows')

    const isShowsSearch = searchOption === "shows"

    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
       setInput(e.target.value)
    }
    const onSearch = async () => {
        const result = await apiGet(`/search/${searchOption}?q=${input}`)
        setMovies(result)
    }
    const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
        if (e.keyCode === 13) {
            onSearch()
        }
    }
    const onRadioChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setSearchOption(e.target.value)

    }
    const renderMovies = () => {

        if (movies && movies.length === 0) {
            return <div>no result</div>
        }
        if (movies && movies.length > 0) {
            if (movies[0].show) {
                return <div>
                      {movies.map(movie => <div key={movie.show.id}>{movie.show.name }</div>)}
                </div>
            }

            if (movies[0].person) {
                    return <div>
                      {movies.map(movie => <div key={movie?.person.id}>{movie.person.name }</div>)}
                </div>
            }
        }
        return null
    }

  return (
      <div>
          <input type="text" onChange={onInputChange} value={input} onKeyDown={onKeyDown} />
          <div>
              <label htmlFor="shows-search">shows <input id='shows-search' type="radio" value="shows" checked={isShowsSearch} onChange={onRadioChange} /></label>
              <label htmlFor="actors-search">actors <input id='actors-search' type="radio" value="people" checked={!isShowsSearch} onChange={onRadioChange} /></label>
          </div>
          <button type='button' onClick={onSearch}>search</button>
          {renderMovies()}
      </div>
  )
}

export default MainPageLayout
