// layout/MainPageLayout.tsx
import React, { useState } from 'react'
import {apiGet} from "misc/config"

type MainLayputProps = {
    children:React.ReactNode
}

type Movie = {
    score: number;
    show: {
        id: number,
        name: string,
        type: string,
        language: string,
        genres: string[],
        rating: {
            average:number
        }
        image: {
            medium: string,
            original:string
        },
        summary:string
    }
}

const MainPageLayout = ({ children }: MainLayputProps) => {
    const [input, setInput] = useState("")
    const [movies,setMovies] =useState<Movie []>([])

    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
       setInput(e.target.value)
    }

    const onSearch = async () => {
        const result = await apiGet(`/search/shows?q=${input}`)
        setMovies(result)

    }

    const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
        if (e.keyCode === 13) {
            onSearch()
        }
    }

    const renderMovies = () => {
        if (movies && movies.length === 0) {
            return <div>no result</div>

        }

        if (movies && movies.length > 0) {
            return <div>
                {movies.map(movie => <div key={movie.show.id}>{movie.show.name }</div>)}
            </div>
        }
        return null
    }

  return (
      <div>
          <input type="text" onChange={onInputChange} value={input}  onKeyDown={onKeyDown} />
          <button type='button' onClick={onSearch}>search</button>
          {renderMovies()}
      </div>
  )
}

export default MainPageLayout
