// components/show/Seasons.tsx
import React from 'react'
import {SeasonList,SeasonsWrapper} from "./Season.style"

type SeasonProp = {
     id:number,
    url:string,
    number:       number,
    name:         string,
    episodeOrder: number,
    premiereDate: Date,
    endDate:      Date,
    webChannel:   null,
    image:        string,
    summary: string,
    length:number

}

type SeasonsProps = {
    seasons :SeasonProp []

}

const Seasons = ({seasons}:SeasonsProps) => {
  return (
     <SeasonsWrapper>
      <p>
        Seasons in total: <span>{seasons.length}</span>
      </p>
      <p>
        Episodes in total:{' '}
        <span>
          {seasons.reduce((acc, season) => acc + season.episodeOrder, 0)}
        </span>
      </p>
      <SeasonList>
        {seasons.map(season => (
          <div key={season.id} className="season-item">
            <div className='left'>
              <p>Season {season.number}</p>
              <p>
                Episodes: <span>{season.episodeOrder}</span>
              </p>
            </div>
            <div className='right'>
              Aired:{' '}
              <span>
                {season.premiereDate} - {season.endDate}
              </span>
            </div>
          </div>
        ))}
      </SeasonList>
    </SeasonsWrapper>
  )
}

export default Seasons
