// components/show/ShowCard.tsx
import React from 'react'
import { Link } from 'react-router-dom'
import notfound from "images/not-found.png"
import { StyledShowCard } from "./ShowCard.styled"
import { ActorProps } from "components/actor/ActorCard"
import { useShows } from "../../misc/custom-hook"
import {Star} from "../../../style"

export type ShowProps = {
    show: {
        id: number,
        name: string,
        type: string,
        language: string,
        genres: string[],
        rating: {
            average:number
        }
        image: {
            medium: string,
            original:string
        },
    summary: string,
    network: {
          name:string
    }
    premiered: string,
    status: string,
    _embedded: {
      cast: ActorProps[]
      seasons: {
        id:number,
    url:string,
    number:       number,
    name:         string,
    episodeOrder: number,
    premiereDate: Date,
    endDate:      Date,
    webChannel:   null,
    image:        string,
    summary:      string,
      }
    }
    }
}

const ShowCard = ({ show: { summary, image, name, id } }: ShowProps) => {
  const { state, dispatch }= useShows()
const summaryAsText = summary
    ? `${summary.split(' ').slice(0, 10).join(' ').replace(/<.+?>/g, "")}...`
    : 'No description';
  const imgSrc = image ? image.medium : notfound
  const isStarred = state.includes(id)

  const onClick = () => {
    if (isStarred) {
     dispatch({type:"REMOVE",payload:id})
    } else {
      dispatch({type:"ADD",payload:id})
   }

  }

  return (
    <StyledShowCard>
      <div className='img-wrapper'>
        <img src={imgSrc} alt="show" />
      </div>

      <h1>{name}</h1>

      <p>{summaryAsText}</p>

      <div className='btns'>
        <Link to={`/show/${id}`}>Read more</Link>
        <button type="button" onClick={onClick}><Star active={isStarred}/></button>
      </div>
    </StyledShowCard>
  );
}

export default ShowCard
